src/test/java/uk/gov/ch/oauth/Oauth2Test.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.ch.oauth;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.SessionKeys;
+import uk.gov.companieshouse.session.model.SignInInfo;
+import uk.gov.companieshouse.session.store.Store;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class Oauth2Test {
+
+    @Mock
+    private Store store;
+    @Mock
+    private Session session;
+
+    private final SignInInfo signInInfo = new SignInInfo();
+
+    @InjectMocks
+    private Oauth2 oauth2;
+
+    @Test
+    @DisplayName("Test that a valid signed in user's session state is correctly altered")
+    public void testInvalidateSessionWhenSignedIn() {
+        final String zxsValue = "0000000001z";
+        signInInfo.setSignedIn(true);
+        when(session.getSignInInfo()).thenReturn(signInInfo);
+        Map<String, Object> data = setUserSessionData(zxsValue);
+        when(session.getData()).thenReturn(data);
+        oauth2.invalidateSession(session, SessionKeys.SIGN_IN_INFO.getKey(), store);
+
+        assertFalse(data.containsKey(SessionKeys.SIGN_IN_INFO.getKey()));
+        verify(store, only()).delete(zxsValue);
+
+    }
+
+    @Test
+    @DisplayName("Test that a not signed in user is unable to alter session state")
```

Comment:
This is a test that a user cannot be signed out twice, not that they cannot change the state in other ways.

Reply:
True. The name and most of the logic of this test was extracted from a previous signOutController test which similarly only tested that the user couldn't sign out if they were already signed out, rather than whether or not they could change the entire state, so I have changed the name of the test to reflect that

Reply:
Checking that a user who is not signed in cannot change other aspects of the state seems more suited to a test of the sign in process right?

Reply:
Honestly? It just seems like a wrong test. Users who are signed in should be able to change state.
What we want is that the only changes to state are the ones we expect in any given method.

Reply:
How would we do that beyond testing every method that is supposed to be exclusive to a signed-in user?

Reply:
Erm no we're not understanding each other. I meant just testing that nothing is changing in the state in each method that is unexpected should be enought for each method.
If the apps that use this allow a signed in user to sign in again, Accounts should deal with that.
If the apps allow a signed out user to sign out that is again Accounts problem.
If the apps allow a user to make changes to their details outside of the normal process, thats probably everyones problem, but we're not able to fix it.